### PR TITLE
Service/nfp:user : stub some functions.

### PR DESCRIPTION
**Pull request by mailwl**
Used by Zelda: BoTW

### CODE REVIEW

File: src/core/hle/service/nfp/nfp.cpp [hunk 1]
```diff
@@ -33,24 +34,87 @@ class IUser final : public ServiceFramework<IUser> {
...
+
         IPC::ResponseBuilder rb{ctx, 2};
         rb.Push(RESULT_SUCCESS);
     }
+
+    void ListDevices(Kernel::HLERequestContext& ctx) {
+        NGLOG_WARNING(Service_NFP, "(STUBBED) called");
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(0);
```
Comment (Hexagon12) [inline src/core/hle/service/nfp/nfp.cpp]: Any idea why we're pushing 0 here? We are basically saying that there's no NFC controllers on the system.

File: src/core/hle/service/nfp/nfp.cpp [hunk 2]
```diff
@@ -33,24 +34,87 @@ class IUser final : public ServiceFramework<IUser> {
...
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(static_cast<u32>(device_state));
+    }
+
+    void GetNpadId(Kernel::HLERequestContext& ctx) {
+        NGLOG_WARNING(Service_NFP, "(STUBBED) called");
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(0);
```
Comment (Hexagon12) [inline src/core/hle/service/nfp/nfp.cpp]: 0 probably isn't a valid ID. 1 would be better.

**Comment by Hexagon12** [comment]
With this, BotW now tries rendering only on multicore, but since #493 is active, you would get the `Shader not linked` error.

**Comment by mailwl** [comment]
how about Kirby? it has tons unimplemented functions, i don't even get to `AttachAvailabilityChangeEvent`

**Comment by mailwl** [comment]
it's not enough, latter - unimplemented Ioctl2

File: src/core/hle/service/nfp/nfp.cpp [hunk 3]
```diff
@@ -33,24 +35,116 @@ class IUser final : public ServiceFramework<IUser> {
...
+        IPC::RequestParser rp{ctx};
+        const u32 array_size = rp.Pop<u32>();
+
+        ctx.WriteBuffer(&device_handle, sizeof(device_handle));
+
+        NGLOG_WARNING(Service_NFP, "(STUBBED) called, array_size={}", array_size);
+
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(1);
```
Comment (Hexagon12) [inline src/core/hle/service/nfp/nfp.cpp]: Please revert this back to 0.

**Pull request merged**